src/embedded.ts

```diff
@@ -165,7 +174,7 @@ export class RoomWidgetClient extends MatrixClient {
         // does *not* (yes, that is the right way around) wait for this event. Let's
         // start sending this, then once this has rolled out, we can change element-web to
         // use waitForIFrameLoad=false and have a widget API that's less racy.
-        widgetApi.sendContentLoaded();
+        if (sendContentLoaded) widgetApi.sendContentLoaded();
```

Comment:
Fwiw I think the idea here is that it could just send this whether or not the other side needed it. There's no reason not to send it, as far as I can see. If the other side isn't waiting for it, it will just ignore it, so this feels like another moving part that can break.

Reply:
I think the place/time where to send the content_loaded event is a "per app" decision. Each app might want to be considered done loading at a different point in time.
Most webapps have their own loading screen. If we have the widgetloading -> the app loading -> the app we have lots of screen changes. In a widget it would be much more elegant (and I think this is also one of the ideas behind the content-loaded event) if the widget loading screen would just be shown longer and the app's loading screen is skipped entirely.

Reply:
Yep, this is why I was thinking it ought to just be the widget itself that sends it (or decides when to send it). This could be a step towards that though, I assume?

Reply:
Yes. Seeing it as a step towards this makes sense. Its still in the js-sdk and not the widget api repo. So a "just works without knowing how the widget api works in detail" is what we want to have here imo. Thats why having the default so that the js-sdk sends it after it has initialized all the jssdk specifc things is good I think.

My opinion in summary: the create fn optional param is a good compromise between hiding most of the widget specifics but also allowing app devs to interveine if they know what they are doing.

Reply:
This being in the js-sdk makes this even weirder IMO: the js-sdk is abstracting away a part of the widget API is... odd. I'm not really suggesting changing it now though, so this is probably fine.

Reply:
For the embedded matroska mode this is still what we want right? The js sdk is wrapping the widget-api and makes it as simple to use as the js sdk itself.